Core/MIPS/IR/IRJit.cpp

```diff
@@ -0,0 +1,430 @@
+		// Won't loop, since hasSetRounding is only ever set to 1.
+		js.lastSetRounding = js.hasSetRounding;
+		cleanSlate = true;
+	}
+
+	// Drat.  The VFPU hit an uneaten prefix at the end of a block.
+	if (js.startDefaultPrefix && js.MayHavePrefix()) {
+		WARN_LOG(JIT, "An uneaten prefix at end of block: %08x", GetCompilerPC() - 4);
+		js.LogPrefix();
+
+		// Let's try that one more time.  We won't get back here because we toggled the value.
+		js.startDefaultPrefix = false;
+		// TODO ARM64: This crashes.
+		//cleanSlate = true;
+	}
+
+	if (cleanSlate) {
+		// Our assumptions are all wrong so it's clean-slate time.
+		ClearCache();
+		Compile(em_address);
+	}
+}
+
+void IRJit::RunLoopUntil(u64 globalticks) {
+	PROFILE_THIS_SCOPE("jit");
+
+	// ApplyRoundingMode(true);
+	// IR Dispatcher
+	
+	FILE *f;
+	int numBlocks = 0;
+	if (numBlocks) {
+		f = fopen("E:\\blockir.txt", "w");
+	}
+
+	while (true) {
+		// RestoreRoundingMode(true);
+		CoreTiming::Advance();
+		// ApplyRoundingMode(true);
+		if (coreState != 0) {
+			break;
+		}
+		while (mips_->downcount >= 0) {
+			u32 inst = Memory::ReadUnchecked_U32(mips_->pc);
+			u32 opcode = inst >> 24;
+			u32 data = inst & 0xFFFFFF;
+			if (opcode == (MIPS_EMUHACK_OPCODE >> 24)) {
+				IRBlock *block = blocks_.GetBlock(data);
+				if (numBlocks > 0) {
+					// ILOG("Run block at %08x : v1=%08x a0=%08x", mips_->pc, mips_->r[MIPS_REG_V1], mips_->r[MIPS_REG_A0]);
+					fprintf(f, "BLOCK : %08x v0: %08x v1: %08x a0: %08x s0: %08x s4: %08x\n", mips_->pc, mips_->r[MIPS_REG_V0], mips_->r[MIPS_REG_V1], mips_->r[MIPS_REG_A0], mips_->r[MIPS_REG_S0], mips_->r[MIPS_REG_S4]);
+					fflush(f);
+					numBlocks--;
+				}
+				mips_->pc = IRInterpret(mips_, block->GetInstructions(), block->GetConstants(), block->GetNumInstructions());
+			} else {
+				if (mips_->pc == 0x0880de94)
+					logBlocks = 10;
+				// RestoreRoundingMode(true);
+				// ILOG("Compile block at %08x : v1=%08x a0=%08x", mips_->pc, mips_->r[MIPS_REG_V1], mips_->r[MIPS_REG_A0]);
+				Compile(mips_->pc);
+				// ApplyRoundingMode(true);
+			}
+		}
+	}
+
+	// RestoreRoundingMode(true);
+}
+
+u32 IRJit::GetCompilerPC() {
+	return js.compilerPC;
+}
+
+MIPSOpcode IRJit::GetOffsetInstruction(int offset) {
+	return Memory::Read_Instruction(GetCompilerPC() + 4 * offset);
+}
+
+void IRJit::DoJit(u32 em_address, IRBlock *b) {
+	js.cancel = false;
+	js.blockStart = mips_->pc;
+	js.compilerPC = mips_->pc;
+	js.lastContinuedPC = 0;
+	js.initialBlockSize = 0;
+	js.nextExit = 0;
+	js.downcountAmount = 0;
+	js.curBlock = nullptr;
+	js.compiling = true;
+	js.inDelaySlot = false;
+	js.PrefixStart();
+	ir.Clear();
+
+	int partialFlushOffset = 0;
+
+	js.numInstructions = 0;
+	while (js.compiling) {
+		MIPSOpcode inst = Memory::Read_Opcode_JIT(GetCompilerPC());
+		js.downcountAmount += MIPSGetInstructionCycleEstimate(inst);
+		MIPSCompileOp(inst, this);
+		js.compilerPC += 4;
+		js.numInstructions++;
+
+		if (ir.GetConstants().size() > 64) {
```

Comment:
Maybe we should make `64` a const up higher.  I wonder if there'd be any measurable gain to using 128... probably not.

-[Unknown]